frontend/src/components/question-creation/question-elements/SaveQuestionEditorButton.tsx

```diff
@@ -0,0 +1,135 @@
+  setShowQuestionEditor: React.Dispatch<React.SetStateAction<boolean>>;
+  setQuestions: React.Dispatch<React.SetStateAction<QuestionElement[][]>>;
+}
+
+const SaveQuestionEditorButton = ({
+  setShowQuestionEditor,
+  setQuestions,
+}: SaveQuestionEditorButtonProps): React.ReactElement => {
+  const questionError =
+    "Please create a question to be associated with this response";
+  const responseError =
+    "Please add at least one response type for this question";
+  const emptyElementError =
+    "Please ensure all fields are filled. If you do not need this component, please delete this component.";
+
+  const {
+    questionElements,
+    setQuestionElements,
+    setShowEditorError,
+  } = useContext(QuestionEditorContext);
+
+  const setElementError = (element: QuestionElement, errorText: string) => {
+    setQuestionElements((prevElements) => {
+      return updatedQuestionElement(
+        element.id,
+        element.data,
+        prevElements,
+        errorText,
+      );
+    });
+  };
+
+  const isQuestionPairError = (elementError: string) => {
+    return elementError === questionError || elementError === responseError;
+  };
+
+  const validateNoQuestionPairErrors = () => {
+    const questionPairs = questionElements.filter(
+      (element) =>
+        element.type === QuestionElementType.QUESTION ||
+        element.type in ResponseElementType,
+    );
+    for (let index = 0; index < questionPairs.length; index += 1) {
+      const element = questionPairs[index];
+      const isExistingError =
+        element.error && !isQuestionPairError(element.error);
+      if (!isExistingError) {
+        if (element.type === QuestionElementType.QUESTION) {
+          const isLastElement = index === questionPairs.length - 1;
+          if (
+            isLastElement ||
+            !(questionPairs[index + 1].type in ResponseElementType)
+          ) {
+            setElementError(element, responseError);
+            return false;
+          }
+        }
+        if (element.type in ResponseElementType) {
+          const isFirstElement = index === 0;
+          if (
+            isFirstElement ||
+            questionPairs[index - 1].type !== QuestionElementType.QUESTION
+          ) {
+            setElementError(element, questionError);
+            return false;
+          }
+        }
+        setElementError(element, "");
+      }
+    }
+    return true;
+  };
+
+  const validateNoEmptyElementErrors = () => {
+    const emptyElement = questionElements.find(
+      (element) => element.data === "",
+    );
+    if (emptyElement) {
+      setElementError(emptyElement, emptyElementError);
+      return false;
+    }
+    return true;
+  };
+
+  const validateNoExistingErrors = () => {
+    return questionElements.every(
+      (element) =>
+        !element.error ||
+        isQuestionPairError(element.error) ||
+        element.error === emptyElementError,
+    );
+  };
+
+  const validateNoEmptyEditorError = () => {
```

Comment:
i think this message is a little confusing bc its expecting specifically a question. maybe change the error message to "Please add at least one question element to the editor before saving" and rename `validateNoEmptyEditorError` -> `validateHasQuestionError` or smth

<img width="724" alt="image" src="https://user-images.githubusercontent.com/69697180/228711615-5f95edf7-3daa-400b-ae18-64908e7a6940.png">

Reply:
also would we be able to remove the error message once a question is added (instead of only on `save`)

Reply:
will do!